src/scenic/core/object_types.py

```diff
@@ -1255,67 +1256,138 @@ def boundingBox(self):
                 mesh=shape.mesh, dimensions=(width, length, height)
-            )
-            return shapeRegion.inradius
+            ).inradius
+
+        # Define a helper function that computes the support of the inradius,
+        # given the sub supports.
+        def inradiusSupport(width_s, length_s, height_s, shape_s):
+            # Unpack the dimension supports (and ignore the shape support)
+            min_width, max_width = width_s
+            min_length, max_length = length_s
+            min_height, max_height = height_s
+
+            if None in [
+                min_width,
+                max_width,
+                min_length,
+                max_length,
+                min_height,
+                max_height,
+            ]:
+                # Can't get a bound on one or more dimensions, abort
+                return None, None
+
+            min_bounds = np.array([min_width, min_length, min_height])
+            max_bounds = np.array([max_width, max_length, max_height])
+
+            # Extract a list of possible shapes
+            if isinstance(self.shape, Shape):
+                shapes = [self.shape]
+            elif isinstance(self.shape, MultiplexerDistribution) and all(
+                isinstance(opt, Shape) for opt in self.shape.options
+            ):
+                shapes = self.shape.options
+            else:
+                # Something we don't recognize, abort
+                return None, None
 
-        # If we havea uniform distribution over shapes and a supportInterval for each dimension,
-        # we can compute a supportInterval for this object's inradius
+            # Get the inradius for each shape with the min and max bounds
+            min_distances = [
+                MeshVolumeRegion(mesh=shape.mesh, dimensions=min_bounds).inradius
+                for shape in shapes
+            ]
+            max_distances = [
+                MeshVolumeRegion(mesh=shape.mesh, dimensions=max_bounds).inradius
+                for shape in shapes
+            ]
 
-        # Define helper class
-        class InradiusHelper:
-            def __init__(self, support):
-                self.support = support
+            distance_range = (min(min_distances), max(max_distances))
 
-            def supportInterval(self):
-                return self.support
+            return distance_range
 
-        # Extract bounds on all dimensions
-        min_width, max_width = supportInterval(width)
-        min_length, max_length = supportInterval(length)
-        min_height, max_height = supportInterval(height)
+        # Return either the inradius or a FunctionDistribution using the above helpers
+        args = toDistribution((self.width, self.length, self.height, self.shape))
+        if not isLazy(args):
+            return inradiusActual(*args)
+        else:
+            return FunctionDistribution(
+                args=args, kwargs={}, func=inradiusActual, support=inradiusSupport
+            )
```

Comment:
Rather than explicitly calling `toDistribution`, `isLazy`, and `FunctionDistribution` here, couldn't you just stick `@distributionFunction(support=inradiusSupport)` on `inradiusActual` and then `return inradiusActual(...)`?

Reply:
Good idea! Much cleaner this way.